openedx/adg/lms/applications/models.py

```diff
@@ -241,3 +242,68 @@ class Meta:
 
     def __str__(self):
         return 'Application {id}, Admin note {note} '.format(id=self.user_application.id, note=self.note)
+
+
+class OpenPreRequisiteCourseManager(models.Manager):
+    """
+    Manager which returns all open pre requisite entries
+    """
+
+    def get_queryset(self):
+        today = datetime.now()
+        return super().get_queryset().filter(
+            course__start_date__lte=today,
+            course__end_date__gte=today
+        ).prefetch_related('course').values_list('course', flat=True)
+
+
+class PrerequisiteCourseGroup(models.Model):
+    """
+    Model for course groups, for multilingual prereq courses
+    """
+
+    name = models.CharField(verbose_name=_('Course group name'), max_length=255, )
+
+    class Meta:
+        app_label = 'applications'
+
+    def prereq_course_count(self):
+        return self.prereq_courses.count()
+
+    def open_prereq_courses_count(self):
+        return self.prereq_courses(manager='open_prereq_course_manager').count()  # pylint: disable=no-member
+
+    def course_keys(self):
+        return self.prereq_courses(manager='open_prereq_course_manager').all()  # pylint: disable=no-member
+
+    @classmethod
+    def non_empty_prereq_course_groups(cls):
```

Comment:
A manager would have made more sense if we need multiple queries based on this specific filter `objects.filter(prereq_courses__isnull=False)`. Since it has only one usage and nothing more is required, a manager seems like an overhed.